PocketKit/Sources/Sync/V3/V3Client.swift

```diff
@@ -0,0 +1,183 @@
+import Foundation
+
+/**
+ Client used to access Pocket's V3 endpoint which is legacy, but still holds some critical pieces of Pocket's API for now.
+ */
+public class V3Client: NSObject, V3ClientProtocol {
+
+    /**
+     V3 Client Error Types
+     */
+    enum Error: Swift.Error {
+        case invalidResponse
+        case invalidSource
+        case unexpectedRedirect
+        case badRequest
+        case invalidCredentials
+        case serverError
+        case unexpectedError
+        case generic(Swift.Error)
+    }
+
+    enum Constants {
+        static let baseURL = URL(
+            string: ProcessInfo.processInfo.environment["POCKET_V3_BASE_URL"] ?? "https://getpocket.com/v3"
+        )!
+    }
+
+    /**
+     Our http session we are keeping to reuse
+     */
+    let urlSession: URLSessionProtocol
+
+    /**
+     The session provider that holds the current session for the requests
+     */
+    let sessionProvider: SessionProvider
+
+    /**
+     Consumer key to use
+     */
+    let consumerKey: String
+
+    public static func createDefault(
+        sessionProvider: SessionProvider,
+        consumerKey: String
+    ) -> V3Client {
+        let urlSession = URLSession(configuration: URLSessionConfiguration.ephemeral)
+        return V3Client(sessionProvider: sessionProvider, consumerKey: consumerKey, urlSession: urlSession)
+    }
+
+    /**
+     Init our V3Client using the current session provider and our consumer key
+     */
+    public init(
+        sessionProvider: SessionProvider,
+        consumerKey: String,
+        urlSession: URLSessionProtocol
+    ) {
+        self.sessionProvider = sessionProvider
+        self.consumerKey = consumerKey
+        self.urlSession = urlSession
+    }
+
+    /**
+     Helper function to execute a V3 request so we can re-use error responses and the type decoding.
+     */
+    internal func executeRequest<T>(
+        request: URLRequest,
+        decodeable: T.Type
```

Comment:
```suggestion
        decodable: T.Type
```

Just to match the Swift protocol naming